eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -322,6 +328,27 @@ object Router {
       case Right(rcu) => updateChannelUpdateSameSideAs(rcu.channelUpdate)
     }
   }
+  case class LocalChannel(channel: Either[PrivateChannel, PublicChannel]) {
```

Comment:
Should we have `PrivateChannel` and `PublicChannel` extend the same `trait`?

Reply:
Right, probably, I don't know why that wasn't done initially.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1590/commits/273dd81593614f3323350c7ce0e3c52933841ca7
The only downside is the methods that return a new instance of `PublicChannel/PrivateChannel` now return that shared abstraction, not sure whether it's an issue or not.